Normalizer/KebabKeysNormalizer.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Normalizer;
+
+use FOS\RestBundle\Normalizer\Exception\NormalizationException;
+
+/**
+ * Normalizes the array by changing its keys from kebab case to camel case.
+ *
+ * @author Oleg Andreyev <oleg.andreyev@live.com>
+ */
+class KebabKeysNormalizer implements ArrayNormalizerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize(array $data)
+    {
+        $this->normalizeArray($data);
+
+        return $data;
+    }
+
+    /**
+     * Normalizes an array.
+     *
+     * @param array &$data
+     *
+     * @throws Exception\NormalizationException
+     */
+    private function normalizeArray(array &$data)
```

Comment:
Why do you pass data by reference if you don't change data by reference at all but use temporary value `$normalizedData`. No need to pass data by reference then but just return it here.

Reply:
it's actually is changed by reference at the end of the method `$data = $normalizedData;`